### PR TITLE
(#1086) Remove UncheckedScalar and update cactoos-matchers

### PR DESCRIPTION
**Pull request by victornoel**
This is for #1086.

- I had to change A LOT of file, but it wasn't possible to do it in multiple steps since the signature of `Assertion` changed.
- I had to reintroduce `TeeInputHasResult` from cactoos-matchers because it was removed from it (see https://github.com/llorllale/cactoos-matchers/issues/68).

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1139](https://github.com/yegor256/cactoos/issues/1139) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1139](https://github.com/yegor256/cactoos/issues/1139) is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/fedcfbcf-88f1-4a9e-b897-e6568d67e358); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/io/TempFolderTest.java [hunk 1]
```diff
@@ -38,13 +38,11 @@
 public final class TempFolderTest {
 
     @Test
-    public void createsDirectory() {
+    public void createsDirectory() throws Exception {
+        final File dir = new TempFolder().value().toFile();
```
Comment (llorllale) [inline src/test/java/org/cactoos/io/TempFolderTest.java]: @victornoel `TempFolder` is meant to be used in a `try-with-resources` block

File: src/test/java/org/cactoos/io/TeeInputHasResult.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import org.cactoos.text.UncheckedText;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+/**
+ * Matcher for the {@link TeeInput}'s results.
+ *
+ * @since 0.42
+ */
+public final class TeeInputHasResult extends TypeSafeMatcher<TeeInput> {
```
Comment (llorllale) [inline src/test/java/org/cactoos/io/TeeInputHasResult.java]: @victornoel please leave a puzzle for substituting this with `InputHasContent`

File: src/test/java/org/cactoos/io/TempFolderTest.java (hunk 1 above)
Comment (victornoel) [reply src/test/java/org/cactoos/io/TempFolderTest.java]: @llorllale ok... I didn't change anything related to that, but I will add it then

**Comment by victornoel** [comment]
@llorllale thanks, it's done

**Comment by victornoel** [comment]
@llorllale I've rebased on master to fix conflicts

**Comment by llorllale** [comment]
@rultor merge

**Comment by victornoel** [comment]
@llorllale I fixed the todo

**Comment by llorllale** [comment]
@rultor merge

**Comment by victornoel** [comment]
@llorllale sorry about that, this time it should be alright.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (9 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Job [#1139](https://github.com/yegor256/cactoos/issues/1139) is not in the agenda of @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), can't inspect

**Comment by 0crat** [comment]
The job [#1139](https://github.com/yegor256/cactoos/issues/1139) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**